backend/services/deviceauth/api/http/api_devauth.go

```diff
@@ -237,17 +237,13 @@ func (i *DevAuthApiHandlers) SearchDevicesV2Handler(c *gin.Context) {
 		return
 	}
 
-	numDevs := len(devs)
-	hasNext := false
-	if int64(numDevs) > perPage {
-		hasNext = true
-		numDevs = int(perPage)
-	}
+	numDevs := int64(len(devs))
+	hasNext := numDevs > perPage
+
```

Comment:
We're returning one more device than requested (hence the test failures).
```suggestion
	if hasMore {
		numDevs---
	}
```